src/streamlink/stream/dash.py

```diff
@@ -196,6 +196,22 @@ def parse_manifest(cls, session, url, **args):
         if not audio:
             audio = [None]
 
+        locale_lang = session.localization.language.alpha2
+        # if the locale is explicitly set, prefer that language over others
+        available_languages = set(map(lambda a: a and a.lang or "n/a", audio))
+
+        if session.localization.explicit and locale_lang in available_languages:
```

Comment:
this should be done like in `hls.py`, you are calling it twice at the moment

https://github.com/streamlink/streamlink/blob/096bc7a486f45201af2dce464f172cf4693186ec/src/streamlink/stream/hls.py#L366

Reply:
true, it's cached once per playlist with the `hls.py`
it you open more playlists it will run again there